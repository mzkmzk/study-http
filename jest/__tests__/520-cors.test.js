const curl = require('../curl/index')

it('简单跨域请求', async () => {
  const { row } = await curl(`curl -vs -H 'Host: study-http.404mzk.com' -H 'Origin: http://www.404mzk.com' http://47.115.81.18:9500/cors-sample`)
  expect(row).toMatchSnapshot()
})


it('复杂跨域请求', async () => {
  const { row } = await curl(`curl -vs -X OPTIONS -H 'Content-Type: application/json' -H  'Origin: http://www.404mzk.com' -H 'Host: study-http.404mzk.com'  http://47.115.81.18:9500/cors-complex`)
  expect(row).toMatchSnapshot()
  const result = await curl(`curl -vs -X POST -H 'Content-Type: application/json' -H  'Origin: http://www.404mzk.com' -H 'Host: study-http.404mzk.com'  http://47.115.81.18:9500/cors-complex`)
  expect(result.row).toMatchSnapshot()
})
