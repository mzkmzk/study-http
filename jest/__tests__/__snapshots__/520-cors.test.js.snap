// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`复杂跨域请求 1`] = `
"OPTIONS /cors-complex HTTP/1.1
User-Agent: curl/7.38.0
Accept: */*
Content-Type: application/json
Origin: http://www.404mzk.com
Host: study-http.404mzk.com

HTTP/1.1 204 No Content
Server: nginx/1.17.10
Date: mock-time
Connection: keep-alive
Access-Control-Max-Age: 1728000
Content-Type: text/plain charset=UTF-8
Content-Length: 0

"
`;

exports[`复杂跨域请求 2`] = `
"POST /cors-complex HTTP/1.1
User-Agent: curl/7.38.0
Accept: */*
Content-Type: application/json
Origin: http://www.404mzk.com
Host: study-http.404mzk.com

HTTP/1.1 200 OK
Server: nginx/1.17.10
Date: mock-time
Content-Type: application/octet-stream
Content-Length: 26
Connection: keep-alive
Access-Control-Allow-Origin: http://www.404mzk.com
Access-Control-Allow-Credentials: true
Access-Control-Allow-Methods: GET, POST, PUT, DELETE, OPTIONS
Access-Control-Allow-Headers: Accept,Authorization,Cache-Control,Content-Type,DNT,If-Modified-Since,Keep-Alive,Origin,User-Agent,X-Requested-With

hello-world complex-cookie"
`;

exports[`简单跨域请求 1`] = `
"GET /cors-sample HTTP/1.1
User-Agent: curl/7.38.0
Accept: */*
Host: study-http.404mzk.com
Origin: http://www.404mzk.com

HTTP/1.1 200 OK
Server: nginx/1.17.10
Date: mock-time
Content-Type: application/octet-stream
Content-Length: 23
Connection: keep-alive
Access-Control-Allow-Origin: http://www.404mzk.com

hello-world cors-sample"
`;
